src/data/zmod/quotient.lean

```diff
@@ -116,18 +115,40 @@ noncomputable def orbit_zpowers_equiv : orbit (zpowers a) b ≃ zmod (minimal_pe
 /-- The orbit `(ℤ • a) +ᵥ b` is a cycle of order `minimal_period ((+ᵥ) a) b`. -/
 noncomputable def _root_.add_action.orbit_zmultiples_equiv
   {α β : Type*} [add_group α] (a : α) [add_action α β] (b : β) :
-  add_action.orbit (add_subgroup.zmultiples a) b ≃ zmod (minimal_period ((+ᵥ) a) b) :=
-(add_action.orbit_equiv_quotient_stabilizer (add_subgroup.zmultiples a) b).trans
-  (add_action.zmultiples_quotient_stabilizer_equiv a b).to_equiv
+  add_action.orbit (zmultiples a) b ≃ zmod (minimal_period ((+ᵥ) a) b) :=
+(add_action.orbit_equiv_quotient_stabilizer (zmultiples a) b).trans
+  (zmultiples_quotient_stabilizer_equiv a b).to_equiv
 
 attribute [to_additive orbit_zmultiples_equiv] orbit_zpowers_equiv
 
 @[to_additive orbit_zmultiples_equiv_symm_apply]
-lemma orbit_zpowers_equiv_symm_apply
-  (k : zmod (minimal_period ((•) a) b)) : (orbit_zpowers_equiv a b).symm k =
-  (⟨a, mem_zpowers a⟩ : zpowers a) ^ (k : ℤ) • ⟨b, mem_orbit_self b⟩ :=
+lemma orbit_zpowers_equiv_symm_apply (k : zmod (minimal_period ((•) a) b)) :
+  (orbit_zpowers_equiv a b).symm k =
+    (⟨a, mem_zpowers a⟩ : zpowers a) ^ (k : ℤ) • ⟨b, mem_orbit_self b⟩ :=
 rfl
 
+lemma orbit_zpowers_equiv_symm_apply' (k : ℤ) : (orbit_zpowers_equiv a b).symm k =
+  (⟨a, mem_zpowers a⟩ : zpowers a) ^ k • ⟨b, mem_orbit_self b⟩ :=
+begin
+  conv_rhs { rw ← int.mod_add_div k (minimal_period ((•) a) b) },
+  rw [zpow_add, mul_smul, orbit_zpowers_equiv_symm_apply, zmod.coe_int_cast, smul_left_cancel_iff,
+      subtype.ext_iff, orbit.coe_smul, eq_comm, zpow_smul_eq_iff_minimal_period_dvd],
+  apply dvd_mul_right,
```

Comment:
We can extract the following lemma:
```lean
@[to_additive] lemma zpow_smul_mod_minimal_period (n : ℤ) :
  a ^ (n % (function.minimal_period ((•) a) b : ℤ)) • b = a ^ n • b :=
by conv_rhs { rw [← int.mod_add_div n (minimal_period ((•) a) b), zpow_add, mul_smul,
    zpow_smul_eq_iff_minimal_period_dvd.mpr (dvd_mul_right _ _)] }
```
Could you place this after `zpow_smul_eq_iff_minimal_period_dvd`? (And the same for `ℕ`-powers?)

Then the proof simplifies to:
```suggestion
  ext,
  rw [orbit_zpowers_equiv_symm_apply, zmod.coe_int_cast],
  exact zpow_smul_mod_minimal_period _ _ _
```